modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cluster/ClientClusterGetRequest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.platform.client.cluster;
+
+import org.apache.ignite.IgniteCluster;
+import org.apache.ignite.binary.BinaryRawReader;
+import org.apache.ignite.internal.processors.platform.client.ClientConnectionContext;
+import org.apache.ignite.internal.processors.platform.client.ClientLongResponse;
+import org.apache.ignite.internal.processors.platform.client.ClientRequest;
+import org.apache.ignite.internal.processors.platform.client.ClientResponse;
+
+/**
+ * Cluster get request.
+ */
+public class ClientClusterGetRequest extends ClientRequest {
+    /**
+     * Constructor.
+     *
+     * @param reader Reader.
+     */
+    public ClientClusterGetRequest(BinaryRawReader reader) {
+        super(reader);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ClientResponse process(ClientConnectionContext ctx) {
+
+        IgniteCluster cluster = ctx.kernalContext().grid().cluster();
+        ClientCluster clientCluster = new ClientCluster(cluster);
+
+        long resId = ctx.resources().put(clientCluster);
```

Comment:
Can you describe this a little bit more?
Do you suggest to get rid of initial GetCluster call or remove any ClientCluster object allocation at all?

I was bearing in mind this type of usage:
`var clusterGroup = ignite.GetCluster().ForDotNet().ForRemotes();`
... later on
`clusterGroup.ForAttrbiute("attr").GetNodes()`

If we do not use a resource allocation, then we need to manage this on a client side, i.e. remember the filters and apply them on a concrete action so that correct clusterGroup will be created on a server side?

Reply:
1. `GetCluster()` does not need anything on the server side in any case, this is my main point
2. `ForDotNet()` and `ForAttribute()` - yes, we can keep the filter sequence on the client side, same way we keep the flag for `ICacheClient.WithKeepBinary`. Let's avoid server-side resource allocation when possible and all related lifetime management complications on both sides.